src/core_modules/capture-core/components/Pages/ViewEvent/EventDetailsSection/EventDetailsSection.component.js

```diff
@@ -111,11 +114,33 @@ class EventDetailsSectionPlain extends Component<Props> {
     render() {
         const {
             classes,
+            programStage: { name, icon },
+            hasName,
+            hasIcon,
         } = this.props;
         return (
             <div className={classes.container}>
                 <ViewEventSection
-                    header={<ViewEventSectionHeader text="Event details" icon={InfoIcon} />}
+                    header={
+                        <ViewEventSectionHeader
+                            text={
+                                hasName && name ? name : i18n.t('Event details')
+                            }
+                            icon={
+                                hasIcon && icon
+                                    ? () => (
```

Comment:
I was wondering why you need to invoke an function here?

Reply:
This is how the `ViewEventSectionHeader` works. It receives a function for the icon. This is an old component that I need to reuse.